src/main/java/org/opentripplanner/model/plan/legreference/ScheduledTransitLegReference.java

```diff
@@ -123,4 +143,60 @@ public ScheduledTransitLeg getLeg(TransitService transitService) {
 
     return leg;
   }
+
+  /**
+   * Return false if the stop id in the reference does not match the actual stop id in the trip
+   * pattern.
+   * Return true in the specific case where the stop ids differ, but belong to the same parent
+   * station.
+   *
+   */
+  private boolean matchReferencedStopInPattern(
+    TripPattern tripPattern,
+    int stopPosition,
+    FeedScopedId stopId,
+    TransitService transitService
+  ) {
+    if (stopId == null) {
+      // this is a legacy reference, skip validation
+      // TODO: remove backward-compatible logic after OTP release 2.6
+      return true;
+    }
+
+    StopLocation stopLocationInPattern = tripPattern.getStops().get(stopPosition);
```

Comment:
Is there some code that makes sure that the pattern still has enough stops so that this doesn't throw an out-of-bounds exception?

Reply:
the out-of-range check is done right before calling this method